clang/test/AST/HLSL/StructuredBuffer-AST.hlsl

```diff
@@ -17,7 +17,7 @@
 // EMPTY-NEXT: FinalAttr 0x{{[0-9A-Fa-f]+}} <<invalid sloc>> Implicit final
 
 // There should be no more occurrances of StructuredBuffer
-// EMPTY-NOT: StructuredBuffer
+// EMPTY-NOT: {{^\W}}StructuredBuffer

```

Comment:
This still doesn't look right.  Did you test this expression for a positive match before using here?  I tried that expression and it does not match what you're looking for.  I confirmed that this expression works though: `{{[^[:alnum:]]}}StructuredBuffer`.

Reply:
By the way, my suggestion of `{{[^W]}}` was to match any character except `W`.  `{{^\W}}` looks for `W` at the beginning of the line: `^` outside of character set matches the beginning of the line, `\W` is just going to match `W`, I think.  It doesn't suppose character classes starting with `\` such as `\W`.  Instead, you have to use the ones like in my last comment.

Here's a source that I think matches what FileCheck supports pretty well:
https://en.wikibooks.org/wiki/Regular_Expressions/POSIX-Extended_Regular_Expressions

Reply:
Thanks Tex! I forgot `^` has different meaning inside and outside of `[]`. I have updated the tests to use `{{[^[:alnum:]]}}`.